TTS/utils/manage.py

```diff
@@ -102,10 +102,10 @@ def download_model(self, model_name):
         output_model_path = os.path.join(output_path, "model_file.pth.tar")
         output_config_path = os.path.join(output_path, "config.json")
         # NOTE : band-aid for removing phoneme support
-        if "needs_phonemizer" in model_item and model_item["needs_phonemizer"]:
```

Comment:
Do you think gruut can replace the phonemizer (python package) for our pre-trained models?

Reply:
I believe it could. The most accurate way would be for me to generate lexicons and train g2p models using eSpeak originally. Based on [this FAQ entry of gnu.org](https://www.gnu.org/licenses/gpl-faq.en.html#WhatCaseIsOutputGPL), I think it would **not** violate the license:

>  The output of a program is not, in general, covered by the copyright on the code of the program. So the license of the code of the program does not apply to the output, whether you pipe it into a file, make a screenshot, screencast, or video.
>
> The exception would be when the program displays a full screen of text and/or art that comes from the program. Then the copyright on that text and/or art covers the output. Programs that output audio, such as video games, would also fit into this exception. 

What do you think?

Reply:
No sure how this technically works, but keeping compatibility with existing (_phoneme trained_) models would really be great 🥇 .

Reply:
@thorstenMueller I'm going to create an eSpeak lexicon for German, and see if it works on your pre-trained model :+1:

Reply:
Thank you @synesthesiam. I really appreciate that. Let me know if I can help you with that (eg. testing).

Reply:
This would be a great experiment going forward. Thx!

Reply:
I've created an eSpeak version of the gruut German model (with g2p), and it appears to work just fine as a drop-in replacement for phonemizer with the `tts_models/de/thorsten/tacotron2-DCA` model!

@thorstenMueller Do you have some "hard" test sentences I can run to verify that the pronunciations match with TTS v0.13.0? I will upload WAV files for you to check, since my ears are not sophisticated enough :confused: 

@erogol Would you prefer I pursue this route instead, and save the use of gruut phonemes for the "direct integer array" method for phonemes we had talked about previously? Specifically, I mean:

* Generate gruut lexicons and g2p models using eSpeak IPA, which should be compatible with existing phonemizer models
* Help later on an external phoneme method for advanced users who want full control of their phoneme set

Reply:
I think it makes sense to follow these two steps incrementally.

It is also great to see gruut works as an I place replacement for the old models.

Reply:
> @thorstenMueller Do you have some "hard" test sentences I can run to verify that the pronunciations match with TTS v0.13.0? I will upload WAV files for you to check, since my ears are not sophisticated enough confused

Hi @synesthesiam. Thanks for your support on that :+1: .
Here are same phrases i thought on with typical german umlauts.

* Erstmal nur ein Satz ohne besondere deutsche Sonderzeichen.
* Ein längerer Satz, mit einem Komma sollte auch funktionieren.
* Es gibt auch Texte die ein Komma haben, aber keine besonderen deutschen Zeichen.
* Ö wie in Österreich oder löten.
* Ä ähnliche wie in Anwärter.
* Überraschung beginnt mit einem Ü.
* Süßkram hat ein ß und ein ü in einem Satz.
* Straße oder Strasse sollte sich einigermaßen gleich anhören.
* René ist ein Name der aus dem französischen stammt.
* Klingt TTS und T T S auch gleich?

Reply:
Thanks, @thorstenMueller here are the results: https://drive.google.com/drive/folders/19PR2hCWsN5FIwlUOhcFAkFK_lqF3XgMl?usp=sharing

The first "TTS" in the last sentence does not come out right. I don't have any kind of [initialism](https://www.merriam-webster.com/dictionary/initialism) detection in gruut. Maybe I should add this?

Reply:
Thanks for providing the samples super fast.
In addition to your mentioned TTS i think that

* é (like in René)
* ß (like in Straße)

are completely missing.
"Classical" german umlauts like ä ö ü seem to work quite well.

Reply:
The german ß and the french é in the 7th and 9th sentence are not recognized.
I think they are  not included in the gruut phoneme list.
I added a quick comparison with the espeak and the new gruut phonemization in my [TTS comparison webpage](https://mbarnig.github.io/TTS-Models-Comparison/).

Reply:
Something is happening to the German ß before it's reaching gruut; I need to double-check the phoneme cleaners used in this model.

The French "é", however is not showing up because my g2p model had none in the training data. So I may need to extend my training set with some loanwords.